src/domain/session/room/RoomViewModel.js

```diff
@@ -197,18 +198,87 @@ export class RoomViewModel extends ViewModel {
         }
     }
     
+    async executeJoinCommand(args) {
+        if (args.length == 1) {
+            let roomName = args[0];
+            try {
+                const roomId = await this._options.client.session.joinRoom(roomName);
+                await (await this._options.client.session.observeRoomStatus(roomId)).waitFor(status => status === RoomStatus.Joined);
```

Comment:
You don't need the double await here.

Reply:
I need an await to get the `this._options.client.session.observeRoomStatus(roomId)` object (because he do have the .waitFor) and an await to wait the waitFor to be ended. 

I already tried without the second await, it throws a Type Error because `this._options.client.session.observeRoomStatus(roomId)` returns a Promise.

Reply:
> Is this code better? :
> 
> ```js
>             const roomId = await this._options.client.session.joinRoom(roomName);
>             const roomStatusObserver = await this._options.client.session.observeRoomStatus(roomId);
>             await roomStatusObserver.waitFor(status => status === RoomStatus.Joined);
> ```


ah yes, you do need the double await since it's not a promise that resolves to another promise. This code looks good! :+1: